src/check/parse.zig

```diff
@@ -1,193 +1,45 @@
 const std = @import("std");
-const tokenize = @import("parse/tokenize.zig");
-const Region = @import("../base/Region.zig");
-
-pub const IR = @import("parse/IR.zig");
-
-pub const Node = struct {
-    tag: Tag,
-    data: Data,
-    region: Region,
-
-    pub const Tag = enum {
-        Unary,
-        Binary,
-        // TODO
-    };
 
-    pub const Data = union {
-        Unary: UnaryOpData,
-        Binary: BinaryOpData,
-        // Add more node data as needed
-    };
-
-    pub const UnaryOpData = struct {
-        // TODO
-    };
+const tokenize = @import("parse/tokenize.zig");
+const TokenIndex = tokenize.TokenIndex;
+const TokenizedBuffer = tokenize.TokenizedBuffer;
+const IR = @import("parse/ir.zig");
+const NodeList = IR.NodeList;
+const Diagnostic = IR.Diagnostic;
+const GenCatData = @import("GenCatData");
+const Parser = @import("parse/parser.zig");
+
+source: []const u8,
+tokens: TokenizedBuffer,
+store: IR.NodeStore,
+errors: []const Diagnostic,
+
+/// Parses a single Roc file.  The returned AST should be deallocated by calling deinit
+/// after its data is used to create the next IR, or at the end of any test.
+pub fn parse(allocator: std.mem.Allocator, source: []const u8) !IR {
```

Comment:
I presume we're already on the same page, but I don't think there's a good reason to be throwing errors here, allocation errors should just be caught with `exitOnOom`